legendary_eureka.py

```diff
@@ -31,9 +33,123 @@ def get_jobs(self):
+            for city_dict in location_response:
+                if city_dict['countryName'] == 'Israel':
+                    city_name = ' '.join(city_dict['label'].split()[:-1])
+                    if city_name.lower() == job_location.lower().strip():
+                        job_location = city_dict['label']
+                        break
+
+            if job_location in [dictionary['label'] for dictionary in location_response]:
+                location_response = [dictionary for dictionary in location_response
+                                     if job_location == dictionary['label']]
+
+                loc_id = location_response[0]['locationId']
+                loc_label = location_response[0]['label']
+
+                # Form data to get job results
+                loc_t = 'C'
+                loc_id = loc_id
+                print("Fetching jobs for location: ", loc_label)
+
+            else:
+                # TODO: add note in print message to look in help command that will show available cities
+                #  (from data/json file) for search.
+                print(f'Job location "{job_location}" not found')
+                return False
+
+        else:  # searching jobs in all Israel
+            print('Printing jobs from all locations in Israel')
+            loc_t = 'N'
+            loc_id = 119
+
+        data = {
+            'clickSource': 'searchBtn',
+            'sc.keyword': job_title,
+            'locT': loc_t,
+            'locId': loc_id,
+            'jobType': job_type
+        }
+
+        response = requests.post(job_listing_url, headers=headers, data=data)
+        soup = BeautifulSoup(response.content, "html.parser")
+
+        job_listings = {}
+        for a in soup.find_all('a', href=True):
+
+            # filtering job links that not containing job title (clickable image/company name/location links)
+            if a['href'].startswith('/partner/') and \
+                    (a.string is not None) and \
+                    (a.string != 'JobInfo'):
+
+                link = a['href']
+                full_link = 'https://www.glassdoor.com' + link
+                job_id = link.split('jobListingId=')[1]
+
+                # avoid possible duplicates by comparing details
+                if (job_location in job_listings) and (job_id not in job_listings[job_location]):
+                    job_listings[job_location].update({job_id: [a.string, full_link]})
+
+                elif (job_location in job_listings) and (a.string not in job_listings[job_location][job_id]):
+                    job_listings[job_location][job_id].append(a.string)
+
+                else:
+                    job_listings[job_location] = {job_id: [a.string, full_link]}
+
+        # printing jobs
+        for val in job_listings.values():
+            for index, job_info in enumerate(val.values()):
+                print(f'#{index + 1} Job Title: {job_info[0]}\nJob URL: {job_info[1]}')
```

Comment:
We don't need to print this to stdout, the function can return a dictionary and then we will create a function that prints  dictionaries